app/Libraries/Session/Store.php

```diff
@@ -3,295 +3,238 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
+declare(strict_types=1);
+
 namespace App\Libraries\Session;
 
 use App\Events\UserSessionEvent;
-use Auth;
-use Illuminate\Support\Str;
+use Illuminate\Redis\Connections\PhpRedisConnection;
+use Illuminate\Support\Arr;
 use Jenssegers\Agent\Agent;
-use LaravelRedis as Redis;
 
 class Store extends \Illuminate\Session\Store
 {
-    const SESSION_ID_LENGTH = 40;
+    private const PREFIX = 'sessions:';
 
-    public static function destroy($userId, ?string $excludedSessionId = null)
+    public static function batchDelete(?int $userId, ?array $ids = null): void
     {
-        if (!static::isUsingRedis()) {
+        $ids ??= static::ids($userId);
+
+        if (empty($ids)) {
             return;
         }
 
-        $keys = static::keys($userId);
-        if ($excludedSessionId !== null) {
-            $keys = array_filter($keys, fn ($key) => $key !== $excludedSessionId);
+        $currentSession = \Session::instance();
+        if (in_array($currentSession->getId(), $ids, true)) {
+            $currentSession->flush();
         }
-        UserSessionEvent::newLogout($userId, $keys)->broadcast();
-        Redis::del(array_merge([static::listKey($userId)], $keys));
-    }
 
-    public static function isUsingRedis()
-    {
-        return config('session.driver') === 'redis';
-    }
-
-    /**
-     * Get the redis key prefix for the given user (excluding cache prefix).
-     *
-     * @return string
-     */
-    public static function keyPrefix($userId)
-    {
-        return 'sessions:'.($userId ?? 'guest');
-    }
-
-    public static function keys($userId)
-    {
-        if (!static::isUsingRedis()) {
-            return [];
+        $redis = self::redis();
+        $redis->del($ids);
+        if ($userId !== null) {
+            $idsForEvent = self::keysForEvent($ids);
+            // Also delete ids that were previously stored with prefix which is
+            // the full redis key just like for event.
+            $redis->srem(self::listKey($userId), ...[...$ids, ...$idsForEvent]);
```

Comment:
just should work `$redis->srem(self::listKey($userId), ...$ids, ...$idsForEvent);`